content-review/impl/urkund/src/main/java/org/sakaiproject/contentreview/urkund/UrkundContentValidator.java

```diff
@@ -0,0 +1,117 @@
+import java.util.Map;
+import java.util.Set;
+import java.util.SortedSet;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.sakaiproject.component.api.ServerConfigurationService;
+import org.sakaiproject.content.api.ContentResource;
+import org.sakaiproject.contentreview.service.ContentReviewService;
+import org.sakaiproject.entity.api.ResourceProperties;
+
+/**
+ * This class contains the implementation of 
+ * {@link ContentReviewService.isAcceptableContent}.
+ *
+ */
+public class UrkundContentValidator {
+	private static final Log log = LogFactory.getLog(UrkundContentValidator.class);
+	
+	private int max_file_size;
+    private static int DEFAULT_MAX_FILE_SIZE = 20971520; //20Mb
+	
+	ContentReviewService contentReviewService = null;
+	public void setContentReviewService(ContentReviewService contentReviewService) {
+		this.contentReviewService = contentReviewService;
+	}
+	
+	ServerConfigurationService serverConfigurationService;
+		public void setServerConfigurationService(ServerConfigurationService serverConfigurationService) {
+		this.serverConfigurationService = serverConfigurationService;
+	}
+	
+	public void init() {
+		max_file_size = serverConfigurationService.getInt("urkund.maxFileSize", DEFAULT_MAX_FILE_SIZE);
+	}
+	
+	public boolean isAcceptableContent(ContentResource resource) {
+		if (resource == null) {
+			return false;
+		}
+		String mime = resource.getContentType();
+		log.debug("Got a content type of " + mime);
+
+		Map<String, SortedSet<String>> acceptableExtensionsToMimeTypes = contentReviewService.getAcceptableExtensionsToMimeTypes();
+		Set<String> acceptableMimeTypes = new HashSet<>();
+		for (SortedSet<String> mimeTypes : acceptableExtensionsToMimeTypes.values()) {
+			acceptableMimeTypes.addAll(mimeTypes);
+		}
+
+		Boolean fileTypeOk = false;
+		if (acceptableMimeTypes.contains(mime)) {
+			fileTypeOk =  true;
+			log.debug("FileType matches a known mime");
+		} else {
+			log.debug("FileType doesn't match a known mime");
+		}
+
+		//as mime's can be tricky check the extensions
+		if (!fileTypeOk) {
+			ResourceProperties resourceProperties = resource.getProperties();
+			String fileName = resourceProperties.getProperty(resourceProperties.getNamePropDisplayName());
+			if (fileName.indexOf(".") > 0) {
+
+				String extension = fileName.substring(fileName.lastIndexOf("."));
+				log.debug("file has an extension of " + extension);
+				Set<String> extensions = acceptableExtensionsToMimeTypes.keySet();
+				if (extensions.contains(extension)) {
+					fileTypeOk = true;
+				}
+				else {
+					// Neither the mime type nor the file extension are accepted
+					fileTypeOk = false;
+				}
+
+			} else {
+				// No extension is not accepted
+				// TODO: Make this configurable
+				fileTypeOk = false;
+			}
+		}
+
+		if(fileTypeOk){
+			if (resource.getContentLength() > max_file_size) {
+				log.debug("File is too big: " + resource.getContentLength());
```

Comment:
To avoid the performance hit of these debug statements, use a parameter:

log.debug("File is too big: {}", resource.getContentLength());